OrbitCore/LinuxEventTracer.cpp

```diff
@@ -211,17 +217,18 @@ void LinuxEventTracerThread::Run(
 
             if (fork.PID() == pid_) {
               // A new thread of the sampled process was spawned.
-              new_thread = fork.TID();
-              // Do not add a new ring buffer to ring_buffers here as we are
-              // already iterating over ring_buffers.
+              thread_forked = fork.TID();
+              // Do not add a new ring buffer to fds_to_ring_buffer here as we
+              // are already iterating over fds_to_ring_buffer.
```

Comment:
This is hard to follow. Maybe have a scheduled_actions list, add a AddThread/RemoveThread actions here and have them invoked after the loop is complete?

Reply:
Done as discussed offline.